### PR TITLE
Refactored the bugsnag plugin. Added notification stages. Added unit …

### PR DESCRIPTION
**Pull request by avareldev**
…tests.

### CODE REVIEW

**Review by migo315 — APPROVED** [review]
OK for me / not tested

File: Factory/ClientFactory.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                    explode(',', $this->config['notifyReleaseStages'])
+                )
+            );
+        }
+
+        if ($this->config['anonymizeIp']) {
+            $client->registerCallback(
+                function (Report $report) {
+                    // replaces the automatically collected IP
+                    $report->setUser(['id' => null]);
```
Comment (mKnoop) [inline Factory/ClientFactory.php:66]: You could use the shopware function for this https://github.com/shopware/shopware/blob/406925fbb6a8e98beb94e71ceca1fd473a6edc31/engine/Shopware/Components/Privacy/IpAnonymizer.php

File: Factory/ClientFactory.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    /**
+     * Creates the bugsnag client
+     *
+     * @return Client
+     */
+    public function create(): Client
+    {
+        if (empty($this->config['apiKey'])) {
```
Comment (mKnoop) [inline Factory/ClientFactory.php:50]: Is the client usable without a apiKey?

Comment (avareldev) [reply Factory/ClientFactory.php:50]: Bugsnag by default tries to get it from the $_ENV variables. Thats what it does if no key is passed.

File: Factory/ClientFactory.php (hunk 1 above)
Comment (avareldev) [reply Factory/ClientFactory.php:66]: I don't think this is necessary here.

**Review by mKnoop — APPROVED** [review]

**Pull request merged**